### PR TITLE
Fix PolygonSelector cursor to temporarily hide during active zoom/pan

### PR DESCRIPTION
**Pull request by Kaustbh**
Issue: #28165

The PolygonSelector now correctly hide the cursor when another widget is active, such as the zoom/pan functionality, ensuring a smooth user experience.. This fix ensures that the cursor is temporarily hidden when another interaction is active.

### CODE REVIEW

**Comment by Kaustbh** [comment]
Could someone please take a moment to review my changes? Any feedback would be greatly appreciated. #28262

**Comment by Kaustbh** [comment]
Please give me some feedback, what is the wrong with this PR.

**Comment by melissawm** [comment]
Hi @Kaustbh - I'm sorry you haven't gotten a review yet, the maintainers for Matplotlib are mostly volunteers and will come around to your PR as soon as possible.

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -4000,11 +4000,18 @@ def onmove(self, event):
         # needs to process the move callback even if there is no button press.
         # _SelectorWidget.onmove include logic to ignore move event if
         # _eventpress is None.
-        if not self.ignore(event):
+
+        # Hide the cursor when interactive zoom/pan is active
+        if self.ignore(event):
+            if not self.canvas.widgetlock.available(self) and self._xys:
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:4006]: I would only move this comment inside instead.
```suggestion
        if self.ignore(event):
            # Hide the cursor when interactive zoom/pan is active
            if not self.canvas.widgetlock.available(self) and self._xys:
```

**Review by QuLogic — APPROVED** [review]
I tried this out and it worked as expected.

**Review by ksunden — APPROVED** [review]

**Comment by ksunden** [comment]
Thank you @Kaustbh, hope to hear from you again! Congrats on your first merged mpl PR!

**Pull request merged**